quinn/src/connection.rs

```diff
@@ -362,6 +363,30 @@ impl Connection {
         }
     }
 
+    /// Accept the next incoming uni-directional stream
+    pub fn accept_uni(&self) -> AcceptUni<'_> {
```

Comment:
So in terms of API design, I wonder if this is the most natural expression of what people need here. The alternative I have in mind would be an `accept()` that yields an enum with `Uni` or `Bi` newtype variants.

Reply:
Interesting thought. I think this form should be our null hypothesis, since it corresponds closely to what's in `main`.

One upside to the current form is that it makes it easy to implement an application protocol that assigns special significance to e.g. the first unidirectional stream. I often use this pattern for a "hello" message, which must be processed before any other work can be done, which would require bidirectional streams to be manually buffered if a single method was used. I would probably redefine my application protocols to require that no bidirectional streams be opened before the first unidirectional stream if we went that way, which is an additional constraint.

Conversely, a combined method makes it easier to define application protocols that handle both stream types uniformly. I haven't encountered any cases that require that myself; do you know of any?

We could actually support both just fine (the semantics of using them simultaneously wouldn't be any weirder than those of calling `accept_uni` concurrently) but it's probably better to keep the interface from growing any more than absolutely necessary.

Reply:
> Conversely, a combined method makes it easier to define application protocols that handle both stream types uniformly. I haven't encountered any cases that require that myself; do you know of any?

I don't know of any. It seems interesting to me that a single `accept()` is (a) maybe more familiar for TCP users and (b) then forces application developers to decide how to deal with streams of the other type, rather than letting them potentially accumulate inside the connection.

Reply:
I think the key here is that they *don't* accumulate. Each stream type is flow-controlled independently, so accepting only a single type safely backpressures the sender. A single method makes this impossible to take advantage of, and I think is therefore a disadvantage, although I agree it's a bit more familiar.

Reply:
That's a good argument!